src/dmd/traits.d

```diff
@@ -1284,12 +1284,26 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         {
             auto s = getDsymbol(o);
             Declaration d;
-            if (!s || (d = s.isDeclaration()) is null)
+            ClassDeclaration c;
+            if (!s || ((d = s.isDeclaration()) is null && (c = s.isClassDeclaration()) is null))
             {
                 e.error("argument to `__traits(getLinkage, %s)` is not a declaration", o.toChars());
                 return new ErrorExp();
```

Comment:
Since linkage applies to structs, functions, and variables (not just classes), it makes sense to include them here, or there'll just be another inevitable bug report.

Reply:
Can I do that as follow-up? Surely moving towards the target is better than not moving towards the target?

Reply:
I need to do investigation to understand the struct case, since they don't have vtables and stuff.

Reply:
`struct`'s don't appear to have a linkage the same way. Let me address that in a follow-up if it needs to be addressed (I'm not so sure)?

Reply:
https://github.com/dlang/dmd/blob/1ea9308b9eeb268a264c559c11dda1b0064ede6b/src/dmd/declaration.d#L281

Reply:
@TurkeyMan do you want to handle structs in this PR or in a follow-up?